File: programs/loops-strings_string-comparison-3.java
String name1 = "Alan Turing";
String name2 = "Ada Lovelace";
int diff = name1.compareTo(name2);
if (diff &lt; 0) {
    System.out.println("name1 comes before name2.");
} else if (diff > 0) {
    System.out.println("name2 comes before name1.");
} else {
    System.out.println("The names are the same.");
}

File: programs/loops-strings_string-comparison-3.cpp
string name1 = "Alan Turing";
string name2 = "Ada Lovelace";
int diff = name1.compare(name2);
if (diff < 0) {
  cout << "name1 comes before name2.\n";
} else if (diff > 0) {
  cout << "name2 comes before name1.\n";
} else {
  cout << "The names are the same.\n";
}